server/src/main/java/com/paypal/selion/proxy/SeLionRemoteProxy.java

```diff
@@ -191,31 +181,64 @@ public TestSession getNewSession(Map<String, Object> requestedCapability) {
                 if (totalSessionsStarted >= maxSessionsAllowed) {
                     startNodeRecycleThread();
                 }
-                appendMsgToCustomLog("Beginning session #" + totalSessionsStarted);
+                proxyLogger.fine("Beginning session #" + totalSessionsStarted + " (" + session.toString() + ")");
             }
+            LOGGER.exiting((session != null) ? session.toString() : null);
             return session;
         }
     }
 
     private void startNodeRecycleThread() {
         if (!nodeRecycleThread.isAlive()) {
-            appendMsgToCustomLog("Spawning NodeRecycleThread to recycle the node " + getId());
+            proxyLogger.fine("Spawning NodeRecycleThread to recycle the node " + getId());
             nodeRecycleThread.start();
         }
     }
 
+    private void stopNodeRecycleThread() {
+        if (nodeRecycleThread != null) {
```

Comment:
Should this test if thread is alive? I don't think  nodeRecycleThread is ever not null.